src/darwin/Framework/CHIP/MTRDevice_XPC.mm

```diff
@@ -144,6 +148,10 @@ - (oneway void)device:(NSNumber *)nodeID stateChanged:(MTRDeviceState)state
 
 - (oneway void)device:(NSNumber *)nodeID receivedAttributeReport:(NSArray<NSDictionary<NSString *, id> *> *)attributeReport
 {
+    if (!MTRInputIsStructurallyValid(nodeID, NSNumber.class) || !MTRInputIsStructurallyValid(attributeReport, MTRInputStructureAttributeReport)) {
```

Comment:
I think stylistically it would be nice to have one
```VerifyOrReturn(MTRInputIsStructurallyValid(nodeID, NSNumber.class));```
per argument or per check, it makes for shorter lines and diffs and easier reading than boolean operators.

Reply:
Did more of that, yes.